R-package/tests/testthat/test_lgb.model.dt.tree.R

```diff
@@ -0,0 +1,135 @@
+NROUNDS <- 10L
+MAX_DEPTH <- 3L
+N <- nrow(iris)
+X <- data.matrix(iris[FEAT])
+FEAT <- colnames(X)
+NCLASS <- nlevels(iris[, 5L])
+
+model_reg <- lgb.train(
+  params = list(
+    objective = "regression"
+    ,  num_threads = .LGB_MAX_THREADS
+    , max.depth = MAX_DEPTH
+  )
+  , data = lgb.Dataset(X, label = iris[, 1L])
+  , verbose = .LGB_VERBOSITY
+  , nrounds = NROUNDS
+)
+
+model_binary <- lgb.train(
+  params = list(
+    objective = "binary"
+    ,  num_threads = .LGB_MAX_THREADS
+    , max.depth = MAX_DEPTH
+  )
+  , data = lgb.Dataset(X, label = iris[, 5L] == "setosa")
+  , verbose = .LGB_VERBOSITY
+  , nrounds = NROUNDS
+)
+
+model_multiclass <- lgb.train(
+  params = list(
+    objective = "multiclass"
+    ,  num_threads = .LGB_MAX_THREADS
+    , max.depth = MAX_DEPTH
+    , num_classes = NCLASS
+  )
+  , data = lgb.Dataset(X, label = as.integer(iris[, 5L]) - 1L)
+  , verbose = .LGB_VERBOSITY
+  , nrounds = NROUNDS
+)
+
+models <- list(reg = model_reg, bin = model_binary, multi = model_multiclass)
+
+for (model_name in names(models)) {
+  model <- models[[model_name]]
+  nrounds <- NROUNDS
+  if (model_name == "multi") {
+    nrounds <- nrounds * NCLASS
+  }
```

Comment:
```suggestion
  expected_n_trees <- NROUNDS
  if (model_name == "multi") {
    expected_n_trees <- NROUNDS * NCLASS
  }
```

Could you please use the language of "trees" instead of "rounds" here, and adjust the tests below appropriately? I think that is slightly easier to understand.

Technically, if you train a multiclass classification model with 3 classes, each "round" of boosting produces 3 trees.

I know this seems pedantic, but this really confused me a lot when I started with LightGBM and I think being a bit clearer about the difference between "rounds" and "trees" might help others looking at this code in the future.

Reply:
Good point, thanks!